pkg/networkgraph/tree/radix.go

```diff
@@ -511,15 +515,7 @@ func (t *nRadixTree) Remove(key string) {
 	}
 
 	node.value = nil
-	parent := node.parent
-	if node.left == nil && node.right == nil {
-		if parent.right == node {
-			parent.right = nil
-		} else {
-			parent.left = nil
-		}
-		removeRecursively(parent)
-	}
+	removeRecursively(node)
```

Comment:
This refactoring is not needed for this PR, but it seems like a good opportunity to make this change.

Reply:
Warning, the behavior is slightly different. In the old version, we had guarantee that one recursive iteration will be executed even if `node.parent == nil`. In the new version, we first check the exit condition (`node.parent == nil`) and only after that we dive deeper.
This may be okay or not depending on the details. Do we have a guarantee here that the `node.parent` is not nil?

Reply:
Oh, now I see that the diff view was hiding the line 513 where we check for `node.parent == nil`. In this case, feel free to ignore my previous comment.

Reply:
Ack